fabric-resource-conditions-api-v1/src/main/java/net/fabricmc/fabric/api/resource/conditions/v1/ConditionJsonProvider.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.resource.conditions.v1;
+
+import com.google.common.base.Preconditions;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.util.Identifier;
+
+/**
+ * A resource condition and its parameters that can be serialized to JSON.
+ */
+public interface ConditionJsonProvider {
```

Comment:
`ConditionSerializer`? Although this is already configured to be for a specific condition...

Reply:
This interface is basically the equivalent of `RecipeJsonProvider` for conditions, that's why I chose that name. :wink:

Reply:
That makes sense then - recipes have serialisers already, so that is just for datagen. I'd clarify in the doc of this class that this is intended for use with the data generation API specifically.

Reply:
Expanded the docs a little.